src/main/kotlin/org/rust/ide/inspections/RustNamingInspection.kt

```diff
@@ -254,7 +254,8 @@ class RustVariableNamingInspection : RustSnakeCaseNamingInspection("Variable") {
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
         object : RustElementVisitor() {
             override fun visitPatBinding(el: RustPatBindingElement) {
-                if (el.parent?.parent !is RustParameterElementImpl) {
+                val parent = el.parent?.parent
+                if (parent is RustLetDeclElement || parent is RustPatTupElement) {
```

Comment:
I think this is not exactly what we want here, consider these two examples:

```Rust
struct S { f: i32 }

let S { f: FALSE_NEGATIVE } = S { f: 92 };

match (a, b) => {
    (None, Some(x)) => "false positive"
} 
```

Ideally we want something more like

```Rust
val pattern = PsiTreeUtil.getTopmostParentOfType(el, RustPatElement::class.java) ?: return
when (pattern.parent) {
    is RustLetDeclElement -> inspect(el.identifier, holder)
}
```

I think it's a high time to introduce 

```
val RustPatBindingElement.rootPattern: RustPatElement?
```

extension. We need something similar over here: https://github.com/intellij-rust/intellij-rust/blob/6a78cae5f629d45608bd9dc79af6ee1605c51a72/src/main/kotlin/org/rust/lang/core/types/visitors/impl/RustTypificationEngine.kt#L229